content/articles/loading-image-with-glide-in-android/index.md

```diff
@@ -0,0 +1,334 @@
+
+![Creating Project](/engineering-education/loading-image-with-glide-in-android/create-project.png)
+
+Click `Finish` and wait for it to build.
+
+### Step 2: Adding Glide to our project
+Add the following dependencies to the app module-level `build.gradle` file:
+
+```gradle
+implementation 'com.github.bumptech.glide:glide:4.12.0'
+annotationProcessor 'com.github.bumptech.glide:compiler:4.12.0'
+```
+
+Add internet permissions to your project.
+
+```xml
+<uses-permission android:name="android.permission.INTERNET"/>
+```
+
+### Step 3: Creating XML layouts
+In this step, we are going to design the XML layout consisting of `Buttons` and an `ImageView`.
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+<ImageView
+    android:id="@+id/imageView"
+    android:layout_width="match_parent"
+    android:layout_height="250dp"
+    android:layout_marginStart="8dp"
+    android:layout_marginTop="8dp"
+    android:layout_marginEnd="8dp"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent"
+    tools:srcCompat="@tools:sample/avatars" />
+<Button
+    android:id="@+id/buttonUrl"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="8dp"
+    android:layout_marginTop="24dp"
+    android:text="URL"
+    app:layout_constraintStart_toStartOf="@+id/imageView"
+    app:layout_constraintTop_toBottomOf="@+id/imageView" />
+<Button
+    android:id="@+id/buttonResize"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="32dp"
+    android:text="Resize Image"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonUrl"
+    app:layout_constraintStart_toEndOf="@+id/buttonUrl"
+    app:layout_constraintTop_toTopOf="@+id/buttonUrl" />
+<Button
+    android:id="@+id/buttonScaling1"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginTop="16dp"
+    android:text="FitCenter"
+    app:layout_constraintStart_toStartOf="@+id/buttonUrl"
+    app:layout_constraintTop_toBottomOf="@+id/buttonUrl" />
+<Button
+    android:id="@+id/buttonScale2"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:layout_marginStart="4dp"
+    android:text="CenterCrop"
+    app:layout_constraintBottom_toBottomOf="@+id/buttonScaling1"
+    app:layout_constraintStart_toEndOf="@+id/buttonScaling1"
+    app:layout_constraintTop_toTopOf="@+id/buttonScaling1" />
```

Comment:
@Collince-Okeyo, these two buttons are a little confusing. Their ids do not match the text or the action they imply.
Consider **refactoring** the ids, for instance, `buttonFitCenter` is more relevant for the `FitCenter` button.